lib/cpp/Kconfig

```
@@ -12,6 +12,26 @@ config CPP
 
 if CPP
 
+config STD_CPP_VERSION
+	int
+	default 1998 if STD_CPP98
+	default 2011 if STD_CPP11
+	default 2014 if STD_CPP14
+	default 2017 if STD_CPP17
+	default 2020 if STD_CPP20 || STD_CPP2A || STD_CPP2B
```

```suggestion
	default 199711L if STD_CPP98
	default 201103L if STD_CPP11
	default 201402L if STD_CPP14
	default 201703L if STD_CPP17
	default 202002L if STD_CPP20 || STD_CPP2A || STD_CPP2B
```
This change makes it comparable to the _cplusplus macro. See 
https://en.cppreference.com/w/cpp/preprocessor/replace#Predefined_macros.

Not sure if we should set 202002 for 2a or 2b, because functionality might be missing.

---

Great suggestion. Thank you